### PR TITLE
[ACTIONS, SOURCES] Productboard new components

### LINKED ISSUE #4096: [ACTIONS, SOURCES]  Productboard new components
**App**
Productboard

**Actions**
Create Note
Create Feature
Get Feature
List Features
Update Feature

**Triggers**
New Note
New Feature
Updated Feature

Requested by user on issue https://github.com/PipedreamHQ/pipedream/issues/4095 along the app integration
API docs at -- https://developer.productboard.com/#section/Introduction
Webhook available for sources -- https://developer.productboard.com/#tag/webhooks

### PR DESCRIPTION
**Pull request by michelle0927**
Resolves #4096 

**New Actions**
- Create Note
- Create Feature
- Get Feature
- List Features
- Update Feature

**New Sources**
- New Feature
- Updated Feature

Source for "New Note" skipped because Productboard doesn't supply a [webhook eventType](https://developer.productboard.com/#operation/postWebhook) for notes, and Productboard's [Notes API](https://developer.productboard.com/#tag/Notes) currently only allows for creating a note.

<a href="https://gitpod.io/#https://github.com/PipedreamHQ/pipedream/pull/4499"></a>

### CODE REVIEW

**Comment by dylburger** [comment]
Thanks for submitting this PR! When we review PRs, we follow the [Pipedream component guidelines](https://pipedream.com/docs/components/guidelines/). If you're not familiar, here's a quick checklist:

- [ ] [Create components to address specific use cases whenever possible](https://pipedream.com/docs/components/guidelines/#component-scope)
- [ ] [Component `key`s should follow the format `app_name_slug-slugified-component-name`](https://pipedream.com/docs/components/guidelines/#component-key-pattern)
- [ ] Components should follow the [standard directory structure](https://pipedream.com/docs/components/guidelines/#folder-structure)
- [ ] [Prefer Node.js client libraries to REST APIs](https://pipedream.com/docs/components/guidelines/#using-apis-vs-client-libraries)
- [ ] When making API requests, [handle pagination](https://pipedream.com/docs/components/guidelines/#pagination) to ensure all data / events are processed
- [ ] [Use `secret` props](https://pipedream.com/docs/components/guidelines/#capturing-sensitive-data) to capture sensitive data
- [ ] Props and methods should be defined in [app files](https://pipedream.com/docs/components/guidelines/#app-files) whenever possible
- [ ] Document methods with [JS Docs](https://pipedream.com/docs/components/guidelines/#methods)
- [ ] [Use `optional` props](https://pipedream.com/docs/components/guidelines/#optional-vs-required-props) whenever possible, and [set a `default` value](https://pipedream.com/docs/components/guidelines/#default-values) where you can
- [ ] Use [async options](https://pipedream.com/docs/components/guidelines/#async-options) to accept user input wherever possible
- [ ] [Create subscription webhooks sources (vs polling sources) whenever possible](https://pipedream.com/docs/components/guidelines/#webhook-vs-polling-sources)
- [ ] [Always emit a `summary` with events](https://pipedream.com/docs/components/guidelines/#emit-a-summary)
- [ ] [Use a built-in deduping strategy wherever possible](https://pipedream.com/docs/components/guidelines/#deduping)
- [ ] When building webhook sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#webhook-sources)
- [ ] When building polling sources, [follow these best practices](https://pipedream.com/docs/components/guidelines/#polling-sources)

File: components/productboard/productboard.app.mjs [hunk 1]
```diff
@@ -0,0 +1,181 @@
...
+      ],
+    },
+  },
+  methods: {
+    _baseUrl() {
+      return "https://api.productboard.com/";
+    },
+    _headers() {
+      return {
+        "Authorization": `Bearer ${this.$auth.api_key}`,
```
Comment (lcaresia) [inline components/productboard/productboard.app.mjs:66]: Move this to a getter method

```suggestion
        "Authorization": `Bearer ${this._apiKey()}`,
```

**Review by lcaresia — CHANGES_REQUESTED** [review]
Just some fews changes!

**Review by lcaresia — APPROVED** [review]
LGTM!

**Comment by DilanAthukorala** [comment]
Verification completed, ready for release.

**Pull request merged**